api/konnect/v1alpha1/konnect_gateway_controlplane_types.go

```diff
@@ -37,6 +37,11 @@ type KonnectGatewayControlPlane struct {
 }
 
 // KonnectGatewayControlPlaneSpec defines the desired state of KonnectGatewayControlPlane.
+// +kubebuilder:validation:XValidation:message="spec.labels must not have more than 40 entries", rule="has(self.labels) ? size(self.labels) <= 40 : true"
```

Comment:
Can this item be replaced by a `MaxLength` comment on `soec.labels`?

Reply:
Unfortunately not as `spec.labels` is defined in an SDK type. We could do that if we replaced the `sdkkonnectcomp.CreateControlPlaneRequest` with a new type defined in this repo. I'm happy to do that, as we do so for other entities already. WDYT?

Reply:
I prefer not to do it now because it will break the implementation in KGO and we do not want to make extra changes.